mir-ci/mir_ci/test_apps_can_run.py

```diff
@@ -16,7 +17,11 @@ class TestAppsCanRun:
         apps.pluma(),
         apps.qterminal(),
     ])
-    async def test_app_can_run(self, server, app) -> None:
-        async with DisplayServer(server) as server:
-            async with server.program(app):
-                time.sleep(short_wait_time)
+    async def test_app_can_run(self, server, app, record_property) -> None:
+        server_instance = DisplayServer(server)
+        program = server_instance.program(app.command, app_type=app.app_type)
```

Comment:
Maybe `Program()` should take a `Dependency` (renamed `App`?), just like `DisplayServer` does?

I know, we're getting into too much of a refactor here, sorry… but then, if `Program` is where we really need `app_type` (to decide on the slice), maybe it's DTRT to propagate it through?

Reply:
- Let's rename `Dependency` to `App` sure
- Making `Program` take an `App` wouldn't be too bad imo, we can do it